packages/api-client-core/spec/repeater.spec.ts

```diff
@@ -0,0 +1,1626 @@
+import {
+    Repeater,
+    RepeaterOverflowError,
+    DroppingBuffer,
+    FixedBuffer,
+    MAX_QUEUE_LENGTH,
+    SlidingBuffer,
+} from "../src/repeater/index.js";
+import { delayPromise } from "./helpers.js";
+
+
+describe("RepeaterBuffer", () => {
```

Comment:
Could you add a little comment here and in the source itself explaining where this code all came from just so the next person who reads it knows that there's an upstream this is copied from?

Reply:
@airhorns Done